daisy/disk.go

```diff
@@ -145,44 +145,44 @@ func (dr *diskRegistry) init() {
 	dr.attachments = map[string]map[string]*diskAttachment{}
 }
 
-func (dr *diskRegistry) deleteFn(res *Resource) dErr {
+func (dr *diskRegistry) deleteFn(res *Resource) DError {
 	m := namedSubexp(diskURLRgx, res.link)
 	err := dr.w.ComputeClient.DeleteDisk(m["project"], m["zone"], m["disk"])
 	if gErr, ok := err.(*googleapi.Error); ok && gErr.Code == http.StatusNotFound {
-		return typedErr(resourceDNEError, err)
+		return typedErr(resourceDNEError, "failed to delete disk", err)
 	}
-	return newErr(err)
+	return newErr("failed to delete disk", err)
```

Comment:
Is it fair to say that here, "failed to delete disk" is practically error type, an enumeration, or error type? This would be the case if safe error messages are always the same when thrown from the same piece of code.

If so, we should consider representing them in a different way than plain strings. This would allow us to remove safeErrMsgs. For example, we could have different error implementation (type) for each type of error. We could use class name as error type.

Reply:
Basically yes, even a int number can represent this info (just maintain a int->string table somewhere). However, we removed safeErrMsgs but added errorCode (or a new class) - we anyways need something new to represent it.

My implementation has 3 extra advantages:
1. Easier to read from log DB.
2. Flexibility - we actually don't want to hide details in all cases. A good example: daisy.Errf("Failed to parse workflow step %v", step.Name). In this case, we actually don't want to hide the step name, because it's not privacy related and it's very useful to identify/categorize the error. The correct code here should be:
daisy.Errf(fmt.Sprintf("Failed to parse workflow step %v", step.Name))
This design gives us the flexibility.
3. The easiest way to add new error type. We don't need to track the largest error code, we don't need to create a new class, we can modify existing error message anytime without side effect.